scripts/get_version_labels.py

```diff
@@ -0,0 +1,76 @@
+"""
+Extract version labels from HathiTrust volume pages.
+"""
+import sys
+import os.path
+import re
+import time
+import datetime
+import requests
+
+
+def get_version_label(htid):
+    """
+    Extract the HathiTrust "version label" from a record's catalog page.
+    Returns the corresponding timestamp, returns None if the HTTP request fails.
+    """
+    re_pattern = r'HT.params.versionLabel = "([^"]+)";'
+    catalog_url = f"https://hdl.handle.net/2027/{htid}"
+    r = requests.get(catalog_url, timeout=5)
+    if r.status_code == requests.codes['ok']:
+        # Extract version_label from response text
+        version_label = re.findall(re_pattern, r.text)
+        if version_label:
+            return version_label[0]
+        else:
+            print(f"Warning: {htid} missing versionLabel!")
+    else:
+        print(f"Warning: bad/unexpected response")
+
+
+def get_version_labels(htids, wait_time=3):
```

Comment:
fwiw, there's a python `ratelimit` library I've used before that can apply rate limiting by adding a decorator to a method. (not needed here, just sharing in case it's of interest)

How did you determine the 3 second delay? Is this a HT rate limit?

Reply:
It was an arbitrary default. That said, there is a [robots.txt](https://babel.hathitrust.org/robots.txt) that specifies a 1 second time delay, so I'll change the default to that.